File: Dot examples/ASP Book Examples/Book Examples/Chapter21/CS/SqlVirtualPathProviderApp/App_Code/SqlVirtualDirectory.cs
using System;
using System.Collections;
using System.Web.Hosting;
using System.Web;

namespace AspNetUnleashed
{

    public class SqlVirtualDirectory : VirtualDirectory
    {

        public SqlVirtualDirectory(string virtualPath) : base(virtualPath) { }


        public string AppPath
        {
            get { return VirtualPathUtility.ToAppRelative(VirtualPath); }
        }


        public override IEnumerable Children
        {
            get {return VirtualFiles.DirectorySelectChildren(VirtualPath); }
        }



        public override IEnumerable Directories
        {
            get { return VirtualFiles.DirectorySelectDirectories(VirtualPath); }
        }

        public override IEnumerable Files
        {
            get { return VirtualFiles.DirectorySelectFiles(VirtualPath); }
        }
    }
}

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/VirtualPathProvider/SqlVirtualDirectory.vb
Imports System
Imports System.Collections
Imports System.Web.Hosting
Imports System.Web

Namespace Superexpert

    Public Class SqlVirtualDirectory
     Inherits VirtualDirectory

        Public Sub New(ByVal virtualPath As String)
            MyBase.New(virtualPath)
        End Sub


        Public ReadOnly Property AppPath() As String
            Get
                 Return VirtualPathUtility.ToAppRelative(VirtualPath)
            End Get
        End Property


        Public Overrides ReadOnly Property Children() As IEnumerable
            Get
                 Return VirtualFiles.DirectorySelectChildren(VirtualPath)
            End Get
        End Property



        Public Overrides ReadOnly Property Directories() As IEnumerable
            Get
                 Return VirtualFiles.DirectorySelectDirectories(VirtualPath)
            End Get
        End Property

        Public Overrides ReadOnly Property Files() As IEnumerable
            Get
                 Return VirtualFiles.DirectorySelectFiles(VirtualPath)
            End Get
        End Property
    End Class
End Namespace

